fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -8,6 +8,38 @@
 from fairlearn.metrics import DemographicParity, BoundedGroupLoss
 from fairlearn.reductions.reductions_estimator import ReductionsEstimator
 from fairlearn.reductions.grid_search import QualityMetric, GridSearchResult
+from fairlearn.reductions.moments.moment import Moment, _REDUCTION_TYPE_CLASSIFICATION
+
+class _GridGenerator:
+    """A generator of a grid of points with a bounded L1 norm.
+    """
+
+    def __init__(self, limit, n_units, dim, neg_allowed, force_L1_norm):
+        # static variables for integer grid accumulation
+        self.dim = dim
+        self.neg_allowed = neg_allowed
+        self.force_L1_norm = force_L1_norm
+
+        # working variables for integer grid accumulation
+        self.entry = np.zeros(self.dim)
+        self.accumulator = []
+        self.accumulate_integer_grid(0, n_units)
```

Comment:
It's a smaller thing, but given that `accumulate_integer_grid` is a recursive method, do we really want to call it in the constructor?

Reply:
I went for a more concise version that allows me to say things like "grid = _GridGenerator(...).grid". But if that rings hard to read, we can refactor.

Reply:
If I've figured out what this is doing correctly, then it's probably not too bad. In general you want minimal logic in a constructor, but it's not like this is doing network calls to create AzureML workspaces (not that I'd have any idea where I got that as an example...). And if this is used once and the grid never needs regeneration, then having a slightly conciser version for use is also desirable.